bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT_DYNAMIC, input);
+
+                } else {
+                    logger.trace("Received input command" + command);
+                    Input input = new Input(((StringType) command).toString());
+                    device.setInput(input);
+                }
+            }
+
+            boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+            boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+            if (isVideoMute || isAudioMute) {
+                if (command == RefreshType.REFRESH) {
+                    // refresh both video and audio mute, as it's one request
+                    MuteQueryResponseValue muteStatus = device.getMuteStatus();
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                            muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                            muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+                } else {
+                    if (isAudioMute) {
+                        logger.trace("Received audio mute command" + command);
+                        boolean muteOn = command == OnOffType.ON;
+                        device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+                    }
+                    if (isVideoMute) {
+                        logger.trace("Received video mute command" + command);
+                        boolean muteOn = command == OnOffType.ON;
+                        device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ResponseException e) {
+            logger.error(e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(PJLinkDeviceConfiguration.class);
+        setupDevice();
+        setupRefreshInterval();
+    }
+
+    private void clearRefreshInterval() {
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    private void handleAuthenticationException(AuthenticationException e) {
+        updateProperty(PJLinkDeviceBindingConstants.PARAMETER_AUTHENTICATION_REQUIRED, new Boolean(true).toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_CONFIGURATION_PENDING, e.getMessage());
```

Comment:
I think ThingStatusDetail.CONFIGURATION_ERROR should be used in this case with an additional message telling the user that he probably forgets to set a password.

Reply:
changed to CONFIGURATION_ERROR;
e.getMessage() will be either "Authentication error, wrong password provided?" or "No password provided, but device requires authentication" in real life